### PR TITLE
fix: tx input > data, add eth_sendTransaction

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <greg.hill@monax.io>

### CODE REVIEW

**Comment by shanejonas** [comment]
hi @gregdhill, the idea with this spec is that you dont use `sendTransaction`, and only use `sendRawTransaction`, many consider `eth_sendTransaction` to be deprecated. we are following EIP 1767 in this regard and leaving them out: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1767.md#rationale

and you can check out more of the discussion around this here: https://ethereum-magicians.org/t/eip-remote-procedure-call-specification/1537/27?u=shanejonas

**Comment by shanejonas** [comment]
we are working towards having `jade-signer-rpc` as the signer, and have a repo similar to this to only cover `signer` methods specification.

heres the repo with its own openrpc.json right now for some context:

https://github.com/etclabscore/jade-signer-rpc

**Comment by BelfordZ** [comment]
in the interest of backwards compatibility, we can consider adding the method, along with the field `deprecated: true` [from OpenRPC](https://spec.open-rpc.org/#method-deprecated) to indicate this desire to no longer use methods that combine account actions and network actions.

thoughts @shanejonas @gregdhill ?

File: openrpc.json [hunk 1]
```diff
@@ -942,6 +942,22 @@
           }
         }
       },
+      {
```
Comment (BelfordZ) [inline openrpc.json]:
```suggestion
      {
        "deprecated": true,
```

File: openrpc.json [hunk 2]
```diff
@@ -942,6 +942,22 @@
...
+        "name": "eth_sendTransaction",
```
Comment (BelfordZ) [inline openrpc.json]:
```suggestion
        "name": "eth_sendTransaction",
        "deprecated": true,
```

**Comment by gregdhill** [comment]
Thanks @shanejonas, I wasn't aware of that thread! I like that idea @BelfordZ, especially as many tools still support these methods - hence why I've added `eth_sign` and `eth_accounts` (with `deprecated: true`). Let me know what you think! Thanks :)

**Comment by BelfordZ** [comment]
@gregdhill if possible, could we merge the addition of eth_sendTransaction with the deprecated field in a single PR, and move discussion of adding the two others to a separate PR?

**Comment by gregdhill** [comment]
@BelfordZ done.

**Comment by BelfordZ** [comment]
Deferring final approval and merging to @shanejonas based on his above concerns.

My views is as such:

1. Don't use sendTransaction. It's a bad pattern - use signTransaction + sendRawTransaction. Use a separate 'signer' service that provides account API. For example: https://github.com/etclabscore/jade-signer-rpc/blob/master/openrpc.json

2. Like it or not, the sendTransaction method does exist, and so if someone *must* use it, we probably should still have some documentation of it. Hence the addition of the 'deprecated' flag.

3. Their should be sufficiently annoying warnings about using an unsafe/deprecated API when using an open-rpc document with deprecated methods.

**Comment by BelfordZ** [comment]
stale, closing.

**Pull request closed without merging**